src/diagnostics/Diagnostic.cpp

```diff
@@ -128,6 +134,25 @@ Diagnostic::ResizeFDiagFAB (amrex::Box local_box, amrex::Box domain, const int l
         domain.grow(Fields::m_slices_nguards);
     }
 
+    if (m_use_custom_size) {
+        const amrex::Real poff_x = GetPosOffset(0, geom, geom.Domain());
+        const amrex::Real poff_y = GetPosOffset(1, geom, geom.Domain());
+        const amrex::Real poff_z = GetPosOffset(2, geom, geom.Domain());
+        amrex::Box cut_domain = domain;
+        cut_domain.setSmall({
+            static_cast<int>(std::round((m_diag_lo[0] - poff_x)/geom.CellSize(0))),
```

Comment:
Could you use something like this to make sure the user doesn't provide a domain that extends beyond the simulation domain (unless this is tested somewhere else)? That way, and if you set `patch_lo` to e.g. `std::numeric_limits<int>::min()`**3 you can also drop the assert (and the user can specify only one)
```suggestion
            stud::max(cut_domain.lo[0], static_cast<int>(std::round((m_diag_lo[0] - poff_x)/geom.CellSize(0)))),
```

Reply:
Calculating a cell index from a position that is infinite might work, but I don’t trust it. 
```
domain &= cut_domain;
local_box &= domain;
```
Here the intersections of the boxes are calculated so the max is not needed.

This feature works with the proper slice diagnostic and with `patch_lo[…] == patch_hi[…]` to get slice diagnostic in any direction at any position.